infrastructure.tf

```diff
@@ -194,3 +194,229 @@ resource "aws_iam_role_policy" "gcp_to_cwl" {
 }
 EOF
 }
+
+////
+// Firehose CWL Log processor
+//
+
+resource "aws_iam_role" "firehose_processor" {
+  name               = "firehose-cwl-log-processor-staging"
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": {
+        "Service": "lambda.amazonaws.com"
+      },
+      "Action": "sts:AssumeRole"
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_iam_role_policy" "firehose_processor" {
+  name   = "firehose-cwl-log-processor-staging"
+  role   = "firehose-cwl-log-processor-staging"
+  policy = <<EOF
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Action": [
+                "logs:CreateLogStream",
+                "logs:PutLogEvents"
+            ],
+            "Resource": "arn:aws:logs:*:*:*"
+        },
+        {
+            "Effect": "Allow",
+            "Action": "logs:CreateLogGroup",
+            "Resource": "arn:aws:logs:*:*:*"
+        },
+        {
+            "Effect": "Allow",
+            "Action": "firehose:*",
+            "Resource": "arn:aws:firehose:*:*:*"
```

Comment:
Might it be worthwhile to limit the firehose to just the domain that the processor needs?

Reply:
yes agreed, will update